### DISCUSSION TITLE
How to create my own task?

### DISCUSSION THREAD

**Discussion by GeorgeOrfanidis** [dmlc/gluon-nlp#1438] [Q&A]
Hello everyone, 

I am trying to use the finetune_classifier.py (https://github.com/dmlc/gluon-nlp/blob/v0.10.x/scripts/bert/finetune_classifier.py) to finetune my own defined task. I have created the training dataset, dev dataset, and testing dataset. However, looking at the command line arguments and specifically on the following:  

```
 parser.add_argument(
        '--task_name',
        type=str,
        choices=['MRPC', 'QNLI', 'RTE', 'STS-B', 'CoLA',
                 'MNLI', 'WNLI', 'SST', 'XNLI', 'LCQMC', 'ChnSentiCorp'],
        help='The name of the task to fine-tune. Choices include MRPC, QQP, '
             'QNLI, RTE, STS-B, CoLA, MNLI, WNLI, SST.')
```
you can specify the above task names. My question is, how can I modify or use other gluon python files to create my own defined task using the training dataset, dev dataset, and testing dataset that I have created? And then use this finetune_classifier.py to finetune bert for the task that I am implementing.

Thank you in advance for any guidance and help.

**Comment by szha** [comment]
If your dataset happens to follow the same format as GLUE benchmarks, you can implement the GlueTask interface:
https://github.com/dmlc/gluon-nlp/blob/e65cd41e872fb174037702591ec86d7c9ba62b60/src/gluonnlp/data/classification.py#L29
It includes a number of metadata fields that are relevant for a dataset.

Here's an example from Quora Question Pair dataset where each sample is a pair of sentences and a label.
https://github.com/dmlc/gluon-nlp/blob/e65cd41e872fb174037702591ec86d7c9ba62b60/src/gluonnlp/data/classification.py#L139-L180

Note that it returns a GlueQQP dataset object but for your custom dataset you don't have to implement it this way.
https://github.com/dmlc/gluon-nlp/blob/e65cd41e872fb174037702591ec86d7c9ba62b60/src/gluonnlp/data/glue.py#L259

Dataset classes are simply list-like containers where it implements `__getitem__` and `__len__`. You can simply use a list of samples for the same purpose. Alternatively, if your file is in TSV/CSV format, you can use the built-in TSVDataset class for loading it.
https://github.com/dmlc/gluon-nlp/blob/e65cd41e872fb174037702591ec86d7c9ba62b60/src/gluonnlp/data/dataset.py#L84

Hope it helps.
